src/code/json-reviver-impl.ts

```diff
@@ -56,6 +59,23 @@ export class JSONReviverImpl implements JSONReviver {
 
     parse(text: string, reviver?: (key: string, value: any) => any): any {
         const reviverCb = reviver ? this._reviverChain.bind(this, reviver) : this._reviver;
-        return JSON.parse(text, reviverCb);
+        const result = JSON.parse(text, reviverCb);
+        if (this.hasUndefined) {
+            this.insertUndefined(result);

```

Comment:
From our code, do you confirm it is better to deserialize in 2 steps ?:
1 - check if undefined marker is present
2 - reparse the whole JSON to apply the "undefined" value

As you may understand, returning immediately "undefined" seems simpler in  #47 ;-)